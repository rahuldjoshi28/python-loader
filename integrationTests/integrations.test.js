const fs = require('fs')
const path = require('path')
const loader = require('../loader')
const terser = require('terser')

jest.mock('terser')

function pythonAsText(testName) {
	return fs.readFileSync(path.join(__dirname, 'python', testName, 'test.py'))
}

function pythonPath(testName) {
	return path.join(__dirname, 'python', testName)
}

const tests = ['simple', 'functions', 'class', 'modules']

describe('integrations', () => {
	beforeEach(() => {
		jest.clearAllMocks()
	})

	it.each(tests)(
		'should work for %s python code and non uglified',
		(testName) => {
			this.context = pythonPath(testName)
			const pythonCode = pythonAsText(testName)
			const loaderWithMockThis = loader.bind(this)
			process.env.UGLIFY = 'false'
			expect(loaderWithMockThis(pythonCode)).toMatchSnapshot()
		}
	)

	it.each(tests)('should work for %s python code and uglified', (testName) => {
		this.context = pythonPath(testName)
		const pythonCode = pythonAsText(testName)
		const loaderWithMockThis = loader.bind(this)

		process.env.UGLIFY = 'false'
		const nonUglifiedCode = loaderWithMockThis(pythonCode)

		expect(terser.minify).not.toHaveBeenCalled()

		process.env.UGLIFY = 'true'
		loaderWithMockThis(pythonCode)

		expect(terser.minify).toHaveBeenLastCalledWith(nonUglifiedCode)
	})
})
