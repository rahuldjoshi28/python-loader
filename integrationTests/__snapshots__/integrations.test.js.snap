// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integrations should work for class python code and non uglified 1`] = `
"module.exports = (function() {
        const range = (start, end, step = 1) => {
  if (end === undefined) {
    return range(0, start);
  }

  let arr = [];

  for (let i = start; i < end; i += step) {
    arr.push(i);
  }

  return arr;
}
        return (function() {
        let \\"class Maths:\\\\n    def add(self, a, b):\\\\n        return a + b\\\\n    def increment(self, i):\\\\n        return self.add(i, 1)\\\\n\\\\nmaths = Maths()\\\\nconsole.log(maths.increment(2))\\"


}

        return { ss Maths:\\\\n    def add }
    })()
    })()"
`;

exports[`integrations should work for functions python code and non uglified 1`] = `
"module.exports = (function() {
        const range = (start, end, step = 1) => {
  if (end === undefined) {
    return range(0, start);
  }

  let arr = [];

  for (let i = start; i < end; i += step) {
    arr.push(i);
  }

  return arr;
}
        return (function() {
        \\"const greet = () => {\\\\n    console.log(\\\\\\"Hello world\\\\\\")\\\\n\\\\ndef add(a, b):\\\\n    return a + b\\\\n\\\\ngreet()\\\\nconsole.log(add(2, 3))\\"


}

        return { greet }
    })()
    })()"
`;

exports[`integrations should work for modules python code and non uglified 1`] = `
"module.exports = (function() {
        const range = (start, end, step = 1) => {
  if (end === undefined) {
    return range(0, start);
  }

  let arr = [];

  for (let i = start; i < end; i += step) {
    arr.push(i);
  }

  return arr;
}
        return (function() {
        l
        return {  }
    })()
    })()"
`;

exports[`integrations should work for simple python code and non uglified 1`] = `
"module.exports = (function() {
        const range = (start, end, step = 1) => {
  if (end === undefined) {
    return range(0, start);
  }

  let arr = [];

  for (let i = start; i < end; i += step) {
    arr.push(i);
  }

  return arr;
}
        return (function() {
        let \\"a = 2\\\\nconsole.log(a)\\"
        return {  }
    })()
    })()"
`;
